libs/custom-paper-handler/src/driver/coders.ts

```diff
@@ -0,0 +1,211 @@
+  returnCode: uint8(),
+  optionalDataLength: uint8(),
+  optionalByte0: uint8(),
+  optionalByte1: uint8(),
+  optionalByte2: uint8(),
+  optionalByte3: uint8(),
+  optionalByte4: uint8(),
+  optionalByte5: uint8(),
+});
+export type PrinterStatusRealTimeExchangeResponse = CoderType<
+  typeof PrinterStatusRealTimeExchangeResponse
+>;
+
+export const RealTimeExchangeResponseWithoutData = message({
+  startOfPacket: literal(0x82),
+  requestId: uint8(),
+  token: literal(TOKEN),
+  returnCode: uint8(),
+  optionalDataLength: literal(0x00),
+});
+export type RealTimeExchangeResponseWithoutData = CoderType<
+  typeof RealTimeExchangeResponseWithoutData
+>;
+
+export const AcknowledgementResponse = uint8();
+export type AcknowledgementResponse = CoderType<typeof AcknowledgementResponse>;
+
+export const ScanResponse = message({
+  signature: literal('IMG'),
+  returnCode: uint8(),
+  cis: uint8(0x00 | 0x01 | 0x02),
+  scan: uint8(),
+  sizeX: uint16(),
+  sizeY: uint16(),
+  status: uint16(),
+  dummy: uint32(), // 4 unused bytes
+});
+export type ScanResponse = CoderType<typeof ScanResponse>;
+
+/**
+ * Command coders
+ */
+export const LoadPaperCommand = literal(0x1c, 0x53, 0x50, 0x4c);
+export type LoadPaperCommand = CoderType<typeof LoadPaperCommand>;
+
+export const ParkPaperCommand = literal(0x1c, 0x53, 0x50, 0x50);
+export type ParkPaperCommand = CoderType<typeof ParkPaperCommand>;
+
+export const EjectPaperCommand = literal(0x1c, 0x53, 0x50, 0x45);
+export type EjectPaperCommand = CoderType<typeof EjectPaperCommand>;
+
+export const PresentPaperAndHoldCommand = literal(0x1c, 0x53, 0x50, 0x46);
+export type PresentPaperAndHoldCommand = CoderType<
+  typeof PresentPaperAndHoldCommand
+>;
+
+export const EjectPaperToBallotCommand = literal(0x1c, 0x53, 0x50, 0x48);
+export type EjectPaperToBallotCommand = CoderType<
+  typeof EjectPaperToBallotCommand
+>;
+
+export const ScannerCalibrationCommand = literal(0x1f, 0x43);
+export type ScannerCalibrationCommand = CoderType<
+  typeof ScannerCalibrationCommand
+>;
+
+export const EnablePrintCommand = literal(0x1f, 0x45);
+export type EnablePrintCommand = CoderType<typeof EnablePrintCommand>;
+
+export const DisablePrintCommand = literal(0x1f, 0x65);
+export type DisablePrintCommand = CoderType<typeof DisablePrintCommand>;
+
+export const ConfigureScannerCommand = message({
+  command: literal(0x1c, 0x53, 0x50, 0x43),
+  optionPaperConfig: uint8(0x00 | 0x01 | 0x02 | 0x03),
+  optionSensorConfig: uint8(0x00 | 0x04),
+  flags: uint8(0x00 | 0x01 | 0x02),
+  cis: literal(0x00), // Unsupported
+  scan: uint8(),
+  dpiX: uint16(undefined, { littleEndian: false }),
+  dpiY: uint16(undefined, { littleEndian: false }),
+  sizeX: uint16(undefined, { littleEndian: false }),
+  sizeY: uint32(undefined, { littleEndian: false }),
+});
+export type ConfigureScannerCommand = CoderType<typeof ConfigureScannerCommand>;
+
+export const GetScannerCapabilityCommand = literal(0x1c, 0x53, 0x43, 0x47);
```

Comment:
https://github.com/votingworks/vxsuite/pull/3545/commits/38507e9663fe76a1b9eb700218c9cdeb757f24b6

I updated hex to string where more readable. Some commands use punctuation so I didn't update those. I also left the first character as hex because IMO it makes it easier to find the reference in the manual, which is indexed alphabetically by hex.